python/tests/operations/join.py

```diff
@@ -0,0 +1,165 @@
+import unittest
+from os import path
+
+import numpy as np
+from numpy.testing import assert_equal
+
+from equistore import Labels
+from equistore.io import load
+from equistore.operations.allclose import allclose_block_raise
+from equistore.operations.join import (  # noqa
+    _join_blocks_along_properties,
+    _join_blocks_along_samples,
+    _join_labels,
```

Comment:
Actually, I am asking why these are private, I did this because we are handling them privately in the other `operations` but equistore has 3 major objects: `TensorMap`, `Labels`, `TensorBlock` which are visible to the user. Why should we not provide public APIs for all of these three instead of only one for the `TensorMap `

Reply:
In my mind, the operation module is only about `TensorMap`. This is the most common use case and allow us to minimize the API surface we need to carefully think about. For example, there could be other way to join labels that do not make sense in the context of joining tensor maps, so making _join_label private means that we don't have to implement these other ways.

I'm not fundamentally against also having functionalities to manipulate blocks & labels, but this will be quite a bit more work that I would rather leave to separate PR and discussions